openhtf/output/web_gui/src/README.md

```diff
@@ -0,0 +1,134 @@
+				|
+				-- example_plugin.html
+				|
+				-- example_service.ts
+```
+
+A plugin with this structure would be able to show up in the frontend and 
+make calls to the backend Tornado server to request information.
+
+### Backend Structure
+
+Everything adding to the backend goes in the __init__.py module! It requires 
+three main parts: 
+
+#### Handler Variable
+
+This tuple is what OpenHTF will use to configure and add your tornado handler. 
+A simple example: 
+```
+handler=([r'/example/', ExampleHandler],
+		 ['host','port'])
+```
+The tuple consists of two parts: 
+- A list containing the regex for the URL you need served, and the name 
+of the handler to serve it to 
+- A list of backend dependencies needed in your handler, an empty list 
+if none needed. Currently, you can get references to the:
+	- station store
+	- host
+	- port
+	- path
+
+#### Configs Variable 
+
+This dictionary specifies your configurations for the frontend portions of 
+the plugin. Make sure these variables match the class and tag you use in your 
+main.ts module; this component will populate the body of your plugin tab.
+Another example: 
+```
+configs={
+	'class':'ExamplePlugin',
+	'tag':'example-plugin', 
+	'label':'example', 
+	'icon':'pets'}
+```
+- label: The label for the tab for your plugin
+- icon: Optional, but refers to the icon from Material Icons that you want
+on your tab. (replace any spaces with underscores)
+
+#### Tornado Handler
+
+If you need to pull information from the backend, then you need to create a 
+Tornado Handler to service it. Some required functions: 
+	- get() - Will return data requested when URL is called. Can take 
+	arguments that refer to capture groups specified in the regex URL section 
+	of the handler variable. 
+	- initialize() - Will initialize any variables you request from the backend. 
+	Arguments should match the dependency list in the handler variable; function 
+	not needed if dependency list is empty.
+
+### Frontend Structure
+
+All frontend assets MUST be in a folder labeled 'plugin' in order to be built
+into OpenHTF. In this folder, your main angular component class MUST be in 
+a module called main.ts. The class and tag in the main.ts must match the class 
+and tag fields in the config variable. This class must also take the following
+inputs in order to be built correctly; whether they are used is up to you:
+- currentMillis: number; 
+- stationInfo: any;
+- currentMillisJob: number;
+- tests: any[];
+
+#### SOME NOTES FOR FRONTEND ASSETS
+
+- For references to styleUrls and templateUrl in the Component wrapper, use the 
+url 'plugin/(NAME OF PACKAGE)/(NAME OF FILE)', 
+i.e, '/plugin/example_plugin/example_plugin.html'
+- For imports of html or css scripts, use urls relative to the plugin folder 
+in your package
+
+## Building the Plugin
+
+Once the contents are finished, it needs to be added to OpenHTF and then built 
+before it can be used. Note: If you are using a virtualenv, this requires npm 
```

Comment:
Is this a hard requirement? Is there any chance we can make the plugin provide the data at runtime?